### DISCUSSION TITLE
CRSF Telemetry

### DISCUSSION THREAD

**Discussion by bayen23** [stronnag/mwptools#134] [Q&A]
Hello,

Is it possible to have all telemetry data from CRSF into mwptools on Telemetry status or other view ?
I have Elapsed, RX bytes, ... Messages but not the data.

Thanks

**Comment by stronnag** [comment]
It would be possible to add a window with more information, but what's the use case?

* If you want to analyse CRSF, without additional visualisation, you can [record the data]( https://github.com/stronnag/mwptools/tree/master/src/samples/mwp-serial-cap) and then [dump out all the data]( https://github.com/stronnag/mwptools/tree/master/src/samples/crsf). 
* If you wanted to see the fields in real time, it would be possible to add serial device reading to the [standalone CRSF parser](https://github.com/stronnag/mwptools/tree/master/src/samples/crsf)

**Reply by bayen23** [reply]
I don't have a specific use case in mind. I fly with a homemade FPV plane associated with a TX16s and an ELRS receiver. I find the idea of ​​your application very interesting. But I was surprised not to have the Telemetry data displayed. And I think that with the aim of monitoring and analyzing a flight, this data seems interesting to me.
How is it possible to add this information in a window ?
How add serial device reading to the standalone CRSF parser ?
Thanks

**Reply by bayen23** [reply]
No answer ?

**Comment by stronnag** [comment]
Should you have had the courtesy to check the recent commits to the repo, you would see that the `crsfparser` already accepts data from standard input, enabling serial, TCP/IP etc. etc.

Please also note this application is a mission planner for INAV and MultiWii; that it understands CRSF is a side-effect, not a requirement and its display of telemetry data is visual / geo-spatial rather than tabular.

Please also note that THIS SOFTWARE IS PROVIDED “AS IS” and there is zero obligation for me to answer hypothetical questions within any time-frame.

**Reply by bayen23** [reply]
Thank you for your response and my apologies for this last message.
